datafusion/core/tests/parquet/row_group_pruning.rs

```diff
@@ -285,105 +285,187 @@ async fn prune_disabled() {
     );
 }
 
-#[tokio::test]
-async fn prune_int32_lt() {
-    RowGroupPruningTest::new()
-        .with_scenario(Scenario::Int32)
-        .with_query("SELECT * FROM t where i < 1")
-        .with_expected_errors(Some(0))
-        .with_matched_by_stats(Some(3))
-        .with_pruned_by_stats(Some(1))
-        .with_matched_by_bloom_filter(Some(0))
-        .with_pruned_by_bloom_filter(Some(0))
-        .with_expected_rows(11)
-        .test_row_group_prune()
-        .await;
-
-    // result of sql "SELECT * FROM t where i < 1" is same as
-    // "SELECT * FROM t where -i > -1"
-    RowGroupPruningTest::new()
-        .with_scenario(Scenario::Int32)
-        .with_query("SELECT * FROM t where -i > -1")
-        .with_expected_errors(Some(0))
-        .with_matched_by_stats(Some(3))
-        .with_pruned_by_stats(Some(1))
-        .with_matched_by_bloom_filter(Some(0))
-        .with_pruned_by_bloom_filter(Some(0))
-        .with_expected_rows(11)
-        .test_row_group_prune()
-        .await;
-}
-
-#[tokio::test]
-async fn prune_int32_eq() {
-    RowGroupPruningTest::new()
-        .with_scenario(Scenario::Int32)
-        .with_query("SELECT * FROM t where i = 1")
-        .with_expected_errors(Some(0))
-        .with_matched_by_stats(Some(1))
-        .with_pruned_by_stats(Some(3))
-        .with_matched_by_bloom_filter(Some(1))
-        .with_pruned_by_bloom_filter(Some(0))
-        .with_expected_rows(1)
-        .test_row_group_prune()
-        .await;
-}
-#[tokio::test]
-async fn prune_int32_scalar_fun_and_eq() {
-    RowGroupPruningTest::new()
-        .with_scenario(Scenario::Int32)
-        .with_query("SELECT * FROM t where i = 1")
-        .with_expected_errors(Some(0))
-        .with_matched_by_stats(Some(1))
-        .with_pruned_by_stats(Some(3))
-        .with_matched_by_bloom_filter(Some(1))
-        .with_pruned_by_bloom_filter(Some(0))
-        .with_expected_rows(1)
-        .test_row_group_prune()
-        .await;
-}
-
-#[tokio::test]
-async fn prune_int32_scalar_fun() {
-    RowGroupPruningTest::new()
-        .with_scenario(Scenario::Int32)
-        .with_query("SELECT * FROM t where abs(i) = 1")
-        .with_expected_errors(Some(0))
-        .with_matched_by_stats(Some(0))
-        .with_pruned_by_stats(Some(0))
-        .with_matched_by_bloom_filter(Some(0))
-        .with_pruned_by_bloom_filter(Some(0))
-        .with_expected_rows(3)
-        .test_row_group_prune()
-        .await;
+macro_rules! int_tests {
+    ($bits:expr, correct_bloom_filters: $correct_bloom_filters:expr) => {
```

Comment:
It would be nice to add documentation about what these parameters mean, especially correct_bloom_filters